### PR TITLE
Adds selection mixin

### PR DESCRIPTION
**Pull request by bhough**
Adds selection mixin.

Also addresses issues in #33 and #40.

### CODE REVIEW

File: src/mixins/selection.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *   'section::-moz-selection': {
+ *     'background-color':'blue',
+ *   },
+ *   'section::selection': {
+ *     'background-color': 'blue',
+ *   }
+ * }
+ */
+
+function selection(styles: Object, parent: string = '') {
```
Comment (bhough) [inline src/mixins/selection.js]: I lean towards no on this one. Unlike other pseudo selectors `::selection` is allowed and often the most common use case, where other pseudos need a parent to function.

File: src/mixins/selection.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * // Styles as object usage
+ * const styles = {
+ *   ...selection({
+ *     'background': 'blue'
+ *   }, 'section')
+ * }
+ *
+ * // styled-components usage
+ * const div = styled.div`
+ *   ${selection({'background': 'blue'}, 'section')}
```
Comment (bhough) [inline src/mixins/selection.js]: My flow-fu isn't super strong. Setting `styles` to `Object|Array` throws this error `Incorrect number of type parameters (expected 1)`. Is there a specific syntax beyond `any` when you want a param to be one of two types?

Comment (mxstbr) [inline src/mixins/selection.js]: Uhh, I don't actually know. It might need to be `Object|Array<any>`? Have you tried that? /cc @relekang

Comment (mxstbr) [inline src/mixins/selection.js]: Just realized that this won't work anyway, so this is good to go!

**Pull request merged**